pkg/services/sqlstore/migrations/ualert/alert_rule.go

```diff
@@ -108,8 +111,17 @@ func addMigrationInfo(da *dashAlert) (map[string]string, map[string]string) {
 
 func (m *migration) makeAlertRule(l log.Logger, cond condition, da dashAlert, folderUID string) (*alertRule, error) {
 	lbls, annotations := addMigrationInfo(&da)
-	annotations["message"] = da.Message
-	var err error
+	tokens, err := tokenizeTmpl(da.Message)
+	if err != nil {
```

Comment:
If we make the changes to retry failed variable parsings as literals, we can safely "log and ignore" errors from the tokenizer (or if this is unpalatable, we can define an error type that we shouldn't ignore but failed variable parsings shouldn't be one of them).

Then if any of the tokens are variables, we use `tokensToTmpl` else we default to `da.Message `.